assets/dev/js/editor/components/template-library/manager.js

```diff
@@ -640,6 +645,26 @@ const TemplateLibraryManager = function() {
 			.setMessage( errorMessage )
 			.show();
 	};
+
+	this.onSelectSourceFilterChange = function( event ) {
+		const select = event.currentTarget,
+			filterName = select.dataset.elementorFilter,
+			templatesSource = select.value;
+
+		elementor.templates.setSourceSelection( templatesSource );
+		elementor.templates.setFilter( filterName, templatesSource, true );
+
+		if ( 'cloud' === templatesSource && ! elementor.config.library_connect.is_connected ) {
+			self.layout.showCloudConnectView();
+			return;
+		}
```

Comment:
Interjecting here so we can avoid an unnecessary AJAX call to `?source=cloud` when the source dropdown is changed.